### PR TITLE
Add example in vignettes for customize label in boxly

### PR DESCRIPTION
**Pull request by wangchen46**
Add new .Rmd file "hover-libeler" int vignettes for examples on how to customize hovering label in boxly.

### CODE REVIEW

File: vignettes/hover-label.Rmd [hunk 1]
```diff
@@ -0,0 +1,167 @@
+---
+title: "Boxly Hover Label"
```
Comment (LittleBeannie) [inline vignettes/hover-label.Rmd:2]: Can we enrich the title something like "Customize hover labels in interactive box plots"?

File: vignettes/hover-label.Rmd [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+- Hover label of outlier is to display the information for the outlier out of box.
+
+  For example: Subject ID, Change from Baseline and etc.
+  
+  
+# Hover label in function `prepare_boxly()`:
+
+**hover_var_outlier**: A character vector of hover variables for outlier.
+
+```
```
Comment (LittleBeannie) [inline vignettes/hover-label.Rmd:44]: Is it better to replace ``` with ```{r, eval=FALSE} ? If you agree, please check out the entire vignette for consistency.

File: vignettes/hover-label.Rmd [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+prepare_boxly(
+  meta,
+  hover_var_outlier = c("USUBJID", metalite::collect_adam_mapping(meta, analysis)$y)
```
Comment (LittleBeannie) [inline vignettes/hover-label.Rmd:47]: Is the `hover_var_outlier` always companied with `hover_outlier_label`, as a 1:1 map? If so, probably we should say something here.

**Review by LittleBeannie — COMMENTED** [review]
Very nice job, @wangchen46 ! The vignettes looks super great to me and tutor me about the label custom. As add-on, I only have 3 editorial comments.

File: vignettes/hover-label.Rmd (hunk 1 above)
Comment (wangchen46) [reply vignettes/hover-label.Rmd]: Updated

File: vignettes/hover-label.Rmd (hunk 2 above)
Comment (wangchen46) [reply vignettes/hover-label.Rmd:48]: {r, eval=FALSE} Added

File: vignettes/hover-label.Rmd (hunk 3 above)
Comment (wangchen46) [reply vignettes/hover-label.Rmd]: Added more details to hover_var_outlier including the one-to-one mapping relationship with hover_outlier_label

**Review by nanxstats — CHANGES_REQUESTED** [review]

File: vignettes/hover-label.Rmd [hunk 4]
```diff
@@ -0,0 +1,165 @@
+---
+title: "Customize Hover Labels in Interactive Box Plot"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Boxly Hover Label}
```
[the remark this answers is not in the document]
Comment (wangchen46) [reply vignettes/hover-label.Rmd]: Updated

File: vignettes/hover-label.Rmd [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+
+> This document will guide the developers on how to customize the hover label based on the study needs.
+
+
+# Overview
+
+Hovering is one of the interactive feature for the box plots. When the user hovers the mouse cursor over a specific part of the plot, additional information related to that specific data point or summary statistic is displayed. 
+
+
+Hovering feature in interactive box plots contains two parts:**Hover label of box** and **Hover label of outlier**.
```
[the remark this answers is not in the document]
Comment (wangchen46) [reply vignettes/hover-label.Rmd]: Whitespace added

File: vignettes/hover-label.Rmd [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+  For example: Subject ID, Change from Baseline and etc.
+  
+  
+# Hover label in function `prepare_boxly()`:
+
+**hover_var_outlier**: A character vector of hover variables for outlier.
+
+```{r, eval=FALSE}
+prepare_boxly <- function(meta,
+                          hover_var_outlier = c("USUBJID", metalite::collect_adam_mapping(meta, analysis)$y)
```
[the remark this answers is not in the document]
Comment (wangchen46) [reply vignettes/hover-label.Rmd:46]: Code Reformatted

File: vignettes/hover-label.Rmd [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+**hover_outlier_label**: A character vector of hover label for outlier.
+
+```{r, eval=FALSE}
+boxly <- function(outdata,
+                  hover_summary_var = c("n", "min", "q1", "median", "mean", "q3", "max"),
+                  hover_outlier_label = c("Participant ID", "Parameter value"),
+                  ...)
+```
+
+The preferred display label name in hover label of outlier corresponding to the value from meta objects could be defined as hover_outlier_label. The default label are "Participant ID" showing the "USUBJID" and the "Parameter value" showing the "CHG" (Change from Baseline). The developer could add more information (i.e. Baseline value, Analysis date). Make sure that the number of hover_outlier_label should match hover_var_outlier and the one-to-one mapping relationship between them.
```
[the remark this answers is not in the document]
Comment (wangchen46) [reply vignettes/hover-label.Rmd]: Updated

File: vignettes/hover-label.Rmd [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+
+# Example 1: Add Hover Label of outlier
+
+In this example, we plan to add more hover labels for outliers.
+
+```{r}
+library(boxly)
+```
+
+Step1: Create a list of metadata using [meta_boxly()](https://merck.github.io/boxly/reference/meta_boxly.html). Using Lab data as example.
```
[the remark this answers is not in the document]
Comment (wangchen46) [reply vignettes/hover-label.Rmd]: Removed the link

File: vignettes/hover-label.Rmd [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+                  ...)
+```
+
+The preferred display label name in hover label of outlier corresponding to the value from meta objects could be defined as hover_outlier_label. The default label are "Participant ID" showing the "USUBJID" and the "Parameter value" showing the "CHG" (Change from Baseline). The developer could add more information (i.e. Baseline value, Analysis date). Make sure that the number of hover_outlier_label should match hover_var_outlier and the one-to-one mapping relationship between them.
+
+The descriptive statistics displayed on hover label of box are defined as hover_summary_var. The developer could choose to display only partial descriptive statistics among 
+"n", "min", "q1", "median", "mean", "q3" or "max". The default value are including all these descriptive statistics.
+
+
+# Example 1: Add Hover Label of outlier
```
[the remark this answers is not in the document]
Comment (wangchen46) [reply vignettes/hover-label.Rmd]: Updated

File: vignettes/hover-label.Rmd [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+Step 3: Call [boxly()](https://merck.github.io/boxly/reference/boxly.html) function to create the interactive plot. We only include "n" for number of participant, "q1" for Q1, "median" for Median, "mean" for Mean, "q3" for Q3. We remove the "min" and "max" from the default value.
+
+```{r}
+boxly(outdata,
+  hover_summary_var = c("n", "q1", "median", "mean", "q3")
+)
+```
+Here, you will notice that only number of participant, Q1, mean, median, Q3 are displaying when pointing to the plot in the box.
+
+# Example 3: Cutomize Label of outlier and Hover Label of box
```
[the remark this answers is not in the document]
Comment (wangchen46) [reply vignettes/hover-label.Rmd]: Blank line added

File: vignettes/hover-label.Rmd [hunk 11]
```diff
@@ -0,0 +1,165 @@
...
+
+Step2: Call [prepare_boxly()](https://merck.github.io/boxly/reference/prepare_boxly.html) function to prepare the metadata as required by the user. In this example, we plan to add Baseline Value and Analysis Date, so besides the default "USUBJID" and "CHG"(as y axis label collected from meta mapping object),"BASE" and "ADT" are also included in the hover_var_outlier.
+
+```{r}
+outdata <- prepare_boxly(meta,
+  hover_var_outlier = c("USUBJID", metalite::collect_adam_mapping(meta, meta$plan$analysis)$y, "BASE", "ADT")
+)
+```
+
+Step 3: Call [boxly()](https://merck.github.io/boxly/reference/boxly.html) function to create the interactive plot. In this step, "Base Value" and "Analysis Date" are included in hover_outlier_label as displaying label for "BASE" and "ADT" which were defined in hover_var_outlier from previous step.
```
[the remark this answers is not in the document]
Comment (wangchen46) [reply vignettes/hover-label.Rmd]: Updated. Thank you so much!

**Review by nanxstats — APPROVED** [review]
Looking great! I think we should merge.

**Pull request merged**